test/e2e/upgrade_management_cluster_test.go

```diff
@@ -0,0 +1,352 @@
+		ClusterctlBinaryPath:    clusterctlBinaryPath, // use older version of clusterctl to init the management cluster
+		ClusterProxy:            upgradeClusterProxy,
+		ClusterctlConfigPath:    clusterctlConfigPath,
+		CoreProvider:            e2eConfig.GetProviderLatestVersionsByContract(contract, config.ClusterAPIProviderName)[0],
+		BootstrapProviders:      e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmBootstrapProviderName),
+		ControlPlaneProviders:   e2eConfig.GetProviderLatestVersionsByContract(contract, config.KubeadmControlPlaneProviderName),
+		InfrastructureProviders: e2eConfig.GetProviderLatestVersionsByContract(contract, e2eConfig.InfrastructureProviders()...),
+		LogFolder:               filepath.Join(artifactFolder, "clusters", clusterName),
+	}, e2eConfig.GetIntervals(specName, "wait-controllers")...)
+
+	/*-------------------------------------------------*
+	| Pivote to run ironic/BMO and resources on target |
+	*--------------------------------------------------*/
+	By("Start pivoting")
+	By("Remove Ironic containers from the source cluster")
+	ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
+	if ephemeralCluster == KIND {
+		removeIronicContainers()
+	} else {
+		removeIronicDeployment()
+	}
+
+	By("Create Ironic namespace")
+	upgradeClusterClientSet := upgradeClusterProxy.GetClientSet()
+	ironicNamespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: os.Getenv("IRONIC_NAMESPACE"),
+		},
+	}
+	_, err = upgradeClusterClientSet.CoreV1().Namespaces().Create(ctx, ironicNamespace, metav1.CreateOptions{})
+	Expect(err).To(BeNil(), "Unable to create the Ironic namespace")
+
+	By("Configure Ironic Configmap")
+	configureIronicConfigmap(true)
+
+	By("Add labels to BMO CRDs")
+	labelBMOCRDs(nil)
+
+	By("Install BMO")
+	installIronicBMO(upgradeClusterProxy, "false", "true")
+
+	By("Install Ironic in the target cluster")
+	installIronicBMO(upgradeClusterProxy, "true", "false")
+
+	By("Restore original BMO configmap")
+	restoreBMOConfigmap()
+	By("Reinstate Ironic Configmap")
+	configureIronicConfigmap(false)
+
+	By("Add labels to BMO CRDs in the target cluster")
+	labelBMOCRDs(upgradeClusterProxy)
+
+	By("Ensure API servers are stable before doing move")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return bootstrapClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "5s", "100ms").Should(BeNil(), "Failed to assert bootstrap API server stability")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return upgradeClusterClient.Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "5s", "100ms").Should(BeNil(), "Failed to assert target API server stability")
+	By("Moving the cluster to self hosted")
+
+	cmd := exec.Command(clusterctlBinaryPath, "move", "--to-kubeconfig", upgradeClusterProxy.GetKubeconfigPath(), "-n", namespace, "-v", "10")
```

Comment:
Could you please remind me why do we have to use the clusterctl binary instead of using the test framework API?

Reply:
Since we are doing the move between v1alpha4 clusters cannot use the framework since it will use v1beta1 clusterctl